test_runner/fixtures/utils.py

```diff
@@ -633,6 +633,11 @@ def allpairs_versions():
     combinations were pre-computed to test all the pairs of the components with
     the different versions.
     """
+    if (os.getenv("COMPATIBILITY_NEON_BIN") is None) or (
+        os.getenv("COMPATIBILITY_POSTGRES_DISTRIB_DIR") is None
+    ):
+        log.info("Compatibility paths are not set, testing only new versions")
+        return {"argnames": "combination", "argvalues": [None]}
```

Comment:
This will also change test names (their parameters part), I don't think it's desirable.
Let's try to skip all combination except all-new using [`marks`](https://docs.pytest.org/en/stable/example/parametrize.html#set-marks-or-test-id-for-individual-parametrized-test) (we might need to switch to using `pytest.param` for that explicitly). 

```python
marks=pytest.mark.skip(reason="COMPATIBILITY_NEON_BIN or COMPATIBILITY_POSTGRES_DISTRIB_DIR is not set")
```